test_framework/terraform/azure/aks/main.tf

```diff
@@ -0,0 +1,60 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=3.0.0"
+    }
+  }
+}
+
+# Configure the Microsoft Azure Provider
+provider "azurerm" {
+  features {}
+
+  client_id                   = var.azure_client_id
+  client_certificate_path     = var.azure_crt_path
+  client_certificate_password = var.azure_crt_password
+  tenant_id                   = var.azure_tenant_id
+  subscription_id             = var.azure_subscription_id
+}
+
+resource "random_string" "random_suffix" {
+  length           = 8
+  special          = false
+  lower            = true
+  upper            = false
+}
+
+resource "azurerm_resource_group" "resource_group" {
+  name     = "${var.test_name}-${random_string.random_suffix.id}-resources"
+  location = "Southeast Asia"
```

Comment:
nit: can be variable instead of hard code for region.

Reply:
Set the region/location as a variable for each managed k8s services means we have to provide options/parameters on the Jenkins pipeline, but currently all types of regression pipelines share the same `Jenkinsfile`, if these parameters have to added, they will be all added in the same `Jenkinsfile`, and it would cause there are redundant parameters only used in some specific pipeline, the parameters initialization could be tedious too. Suggest if we really need it running on different region/location, refactor the `Jenkinsfile` and the setup scripts to make each pipeline more independent.

Reply:
@yangchiu let's create an issue for this to improve this later. I think hard code is not the right way.

Reply:
Created https://github.com/longhorn/longhorn/issues/5960 for this.